google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/GooglePubSubSourceGrpc.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.googlecloud.pubsub
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import akka.actor.ActorSystem
+import akka.stream.{Attributes, Materializer, Outlet, SourceShape}
+import akka.stream.stage.{GraphStage, GraphStageLogic, OutHandler}
+
+import scala.util.{Failure, Success, Try}
+import GooglePubSubSourceGrpc._
+import com.google.pubsub.v1
+
+import scala.collection.JavaConverters._
+import scala.collection.immutable
+
+@akka.annotation.InternalApi
+private final class GooglePubSubSourceGrpc(parallelism: Int, grpcApi: GrpcApi)(
+    implicit as: ActorSystem
+) extends GraphStage[SourceShape[v1.ReceivedMessage]] {
+
+  val out: Outlet[v1.ReceivedMessage] = Outlet("GooglePubSubSourceGrpc.out")
+  override val shape: SourceShape[v1.ReceivedMessage] = SourceShape(out)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    new GraphStageLogic(shape) {
+      private var state: State = Pending
+      private val fetches: AtomicInteger = new AtomicInteger(0)
```

Comment:
No need to make `fetches` an AtomicInteger, the graph stage code won't be executed by more than one thread at a time.